inc/audiences/rest_api/namespace.php

```diff
@@ -0,0 +1,197 @@
+			'type' => 'array',
+			'items' => [
+				'type' => 'object',
+				'properties' => [
+					'name' => [
+						'type' => 'string',
+						'required' => true,
+					],
+					'label' => [
+						'type' => 'string',
+						'required' => true,
+					],
+					'type' => [
+						'type' => 'string',
+						'required' => true,
+					],
+					'data' => [
+						'type' => 'array',
+						'items' => [
+							'type' => 'object',
+							'properties' => [
+								'value' => [ 'type' => 'string' ],
+								'count' => [ 'type' => 'number' ],
+							],
+						],
+					],
+					'stats' => [
+						'type' => 'object',
+						'properties' => [
+							'count' => [ 'type' => 'number' ],
+							'min' => [ 'type' => 'number' ],
+							'max' => [ 'type' => 'number' ],
+							'avg' => [ 'type' => 'number' ],
+							'sum' => [ 'type' => 'number' ],
+						],
+					],
+				],
+			],
+		],
+	] );
+
+	// Fetch an audience size estimate.
+	register_rest_route( 'analytics/v1', 'audiences/estimate', [
+		[
+			'methods' => WP_REST_Server::READABLE,
+			'callback' => function ( WP_REST_Request $request ) : WP_REST_Response {
+				$audience = $request->get_param( 'audience' );
+				$estimate = get_estimate( $audience );
+				return rest_ensure_response( $estimate );
+			},
+			'permissions_callback' => __NAMESPACE__ . '\\permissions',
+			'args' => [
+				'audience' => [
+					'description' => __( 'A URL encoded audience configuration JSON string', 'altis-analytics' ),
+					'required' => true,
+					'type' => 'string',
+					'validate_callback' => function ( $param ) {
+						$audience = json_decode( urldecode( $param ), true );
+
+						if ( json_last_error() ) {
+							return new WP_Error( 'altis_audience_estimate_json_invalid', json_last_error_msg() );
+						}
+
+						// Validate against the audience schema after decoding.
+						return rest_validate_value_from_schema( $audience, get_audience_schema(), 'audience' );
+					},
+					'sanitize_callback' => function ( $param ) {
+						$audience = json_decode( urldecode( $param ), true );
+
+						if ( json_last_error() ) {
+							return new WP_Error( 'altis_audience_estimate_json_invalid', json_last_error_msg() );
+						}
+
+						return $audience;
+					},
+				],
+			],
+		],
+		'schema' => [
+			'type' => 'object',
+			'properties' => [
+				'count' => [ 'type' => 'number' ],
+				'total' => [ 'type' => 'number' ],
+				'histogram' => [
+					'type' => 'array',
+					'items' => [
+						'type' => 'object',
+						'properties' => [
+							'index' => [ 'type' => 'string' ],
+							'count' => [ 'type' => 'number' ],
+						],
+					],
+				],
+			],
+		],
+	] );
+
+	// Handle the audience configuration data retrieval and saving via the REST API.
+	register_rest_field( POST_TYPE, 'audience', [
+		'get_callback' => function ( array $post ) {
+			return get_estimate( $post['id'] );
+		},
+		'update_callback' => function ( $value, WP_Post $post ) {
+			return update_post_meta( $post->ID, 'audience', $value );
```

Comment:
Does `$value` need to be slashed here?

Reply:
Hmm, maybe yeah. I'm not unslashing it so will do that in both places.